src/main/java/org/cbioportal/persistence/ClinicalDataRepository.java

```diff
@@ -65,5 +57,13 @@ List<ClinicalDataCount> fetchClinicalDataCounts(List<String> studyIds, List<Stri
 
     @Cacheable(cacheResolver = "generalRepositoryCacheResolver", condition = "@cacheEnabledConfig.getEnabled()")
     List<ClinicalData> getPatientClinicalDataDetailedToSample(List<String> studyIds, List<String> patientIds,
-            List<String> attributeIds);
+                                                              List<String> attributeIds);
+
+    List<Integer> getVisibleSampleInternalIdsForClinicalTable(List<String> studyIds, List<String> sampleIds,
```

Comment:
Should these be cacheable?

Reply:
@JREastonMarks The data is cached at the level of the StudyViewController:

```
    // Only cache when:
    // 1) the request concerns the entire study
    // 2) no sorting/searching
    // 3) requesting the first page
    @Cacheable(
        cacheResolver = "staticRepositoryCacheOneResolver",
        condition = "@cacheEnabledConfig.getEnabled() && #singleStudyUnfiltered && (#sortBy == null || #sortBy.isEmpty()) && (#searchTerm == null || #searchTerm.isEmpty()) && #pageNumber == 0"
    )
 public ImmutablePair<SampleClinicalDataCollection, Integer> cachedClinicalDataTableData(
        StudyViewFilter interceptedStudyViewFilter, boolean singleStudyUnfiltered, Integer pageNumber, 
        Integer pageSize, String sortBy, String searchTerm, String sortDirection
    ) {
```